flink-ml-lib/src/main/java/org/apache/flink/ml/feature/variancethresholdselector/VarianceThresholdSelector.java

```diff
@@ -0,0 +1,218 @@
+import org.apache.flink.runtime.state.StateSnapshotContext;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Collector;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+/**
+ * An Estimator which implements the VarianceThresholdSelector algorithm. The algorithm removes all
+ * low-variance features. Features with a variance not greater than the threshold will be removed.
+ * The default is to keep all features with non-zero variance, i.e. remove the features that have
+ * the same value in all samples.
+ */
+public class VarianceThresholdSelector
+        implements Estimator<VarianceThresholdSelector, VarianceThresholdSelectorModel>,
+                VarianceThresholdSelectorParams<VarianceThresholdSelector> {
+
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public VarianceThresholdSelector() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public VarianceThresholdSelectorModel fit(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        final String featuresCol = getFeaturesCol();
+        final double varianceThreshold = getVarianceThreshold();
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<DenseVector> inputData =
+                tEnv.toDataStream(inputs[0])
+                        .map(
+                                (MapFunction<Row, DenseVector>)
+                                        value -> ((Vector) value.getField(featuresCol)).toDense());
+        DataStream<DenseVector> varianceValues =
+                inputData
+                        .transform(
+                                "CalculateVarianceOperator",
+                                inputData.getType(),
+                                new VarianceFunctionOperator())
+                        .setParallelism(1);
+        DataStream<VarianceThresholdSelectorModelData> modelData =
+                DataStreamUtils.mapPartition(
+                        varianceValues,
+                        new RichMapPartitionFunction<
+                                DenseVector, VarianceThresholdSelectorModelData>() {
+                            @Override
+                            public void mapPartition(
```

Comment:
With the current implementation, `varianceValues` is guaranteed to have exactly 1 record unless there is an error.

Would it be simpler to use `modelData = varianceValues.map(...)` instead of calling `DataStreamUtils.mapPartition(...)`?